jax/experimental/sparse/bcoo.py

```diff
@@ -572,6 +576,63 @@ def _bcoo_dot_general_abstract_eval(lhs_data, lhs_indices, rhs, *, dimension_num
   out_dtype = jnp.promote_types(lhs_data.dtype, rhs.dtype)
   return core.ShapedArray(out_shape, out_dtype)
 
+_bcoo_dot_general_default_translation_rule = xla.lower_fun(
+    _bcoo_dot_general_impl, multiple_results=False, new_style=True)
+
+def _bcoo_dot_general_cuda_translation_rule(
+    ctx, avals_in, avals_out, lhs_data, lhs_indices, rhs, *, dimension_numbers,
+    lhs_shape):
+
+  (lhs_contract, rhs_contract), _ = dimension_numbers
+  _, lhs_indices_aval, _, = avals_in
+  _, n_sparse, _, _ = _validate_bcoo_indices(lhs_indices_aval, lhs_shape)
+
+  lhs_transpose = False
+  if n_sparse == 1:
+    # Converts lhs to a row vector.
+    col = xops.Collapse(lhs_indices, dimensions=[0, 1])
+    row = xops.Broadcast(xops.Constant(ctx.builder, np.array(0, dtype=int)),
+                         ctx.builder.get_shape(col).dimensions())
+    lhs_shape = (1, lhs_shape[0]) if len(lhs_shape) == 1 else sorted(lhs_shape)
```

Comment:
Under what circumstance is `len(lhs_shape) != 1`? I think `n_dense == n_batch == 0` and `n_sparse == 1` is sufficient to guarantee that `len(lhs_shape) == `.